src/goto-symex/symex_main.cpp

```diff
@@ -409,17 +415,111 @@ goto_symext::get_goto_function(abstract_goto_modelt &goto_model)
   };
 }
 
+messaget::mstreamt &
+goto_symext::print_callstack_entry(const symex_targett::sourcet &source)
+{
+  log.status() << source.function_id
+               << " location number: " << source.pc->location_number;
+
+  return log.status();
+}
+
+void goto_symext::print_symex_steps(statet &state)
+{
+  // If we're showing the route, begin outputting debug info, and don't print
+  // instructions we don't run.
+
+  // We also skip dead instructions as they don't add much to step-based
+  // debugging and if there's no code block at this point.
+  if(
+    !symex_config.show_symex_steps || state.guard.is_false() ||
+    state.source.pc->type == DEAD ||
+    (state.source.pc->code.is_nil() && state.source.pc->type != END_FUNCTION))
+  {
+    return;
+  }
+
+  if(state.source.pc->code.is_not_nil())
+  {
+    auto guard_expression = state.guard.as_expr();
+    std::size_t size = 0;
+    for(auto it = guard_expression.depth_begin();
+        it != guard_expression.depth_end();
+        ++it)
+    {
+      size++;
+    }
+
+    log.status() << "[Guard Size: " << size << "] "
+                 << format(state.source.pc->code);
+
+    if(
+      state.source.pc->source_location.is_not_nil() &&
+      !state.source.pc->source_location.get_java_bytecode_index().empty())
+    {
+      log.status()
+        << " bytecode index: "
+        << state.source.pc->source_location.get_java_bytecode_index();
+    }
+
+    log.status() << messaget::eom;
+  }
+
+  // Print the method we're returning too.
+  const auto &call_stack = state.threads[state.source.thread_nr].call_stack;
+  if(state.source.pc->type == END_FUNCTION)
+  {
+    log.status() << messaget::eom;
+
+    if(call_stack.empty())
+    {
+      log.status() << "No stack!" << messaget::eom;
+    }
+    else
+    {
+      log.status() << "Returning to: ";
+      print_callstack_entry(call_stack.back().calling_location)
+        << messaget::eom;
+    }
+
+    log.status() << messaget::eom;
+  }
+
+  // On a function call print the entire call stack.
+  if(state.source.pc->type == FUNCTION_CALL)
+  {
+    log.status() << messaget::eom;
+
+    if(call_stack.empty())
+    {
+      log.status() << "No stack!" << messaget::eom;
```

Comment:
This makes it sound like something has gone wrong? I think it might be best to just omit this.